src/cfg.rs

```diff
@@ -415,124 +400,123 @@ impl SExpr {
     }
 }
 
-/// Get the root expressions and strip comments.
-fn get_root_exprs(cfg: &str) -> Result<Vec<String>> {
-    let mut open_paren_count = 0;
-    let mut close_paren_count = 0;
-    let mut s_exprs = Vec::new();
-    let mut cur_expr = String::new();
-    for line in cfg.lines() {
-        // remove comments
-        let line = line.split(";;").next().unwrap();
-        for c in line.chars() {
-            if c == '(' {
-                open_paren_count += 1;
-            } else if c == ')' {
-                close_paren_count += 1;
+#[derive(Debug)]
+enum Token<'a> {
+    Open,
+    Close,
+    String(&'a str),
+}
+
+struct Lexer<'a> {
+    s: &'a str,
+    bytes: Bytes<'a>,
+}
+
+fn is_start(b: u8) -> bool {
```

Comment:
We should choose what we want to accept for bare atoms. I think ascii would best but we could also do something like unicode_xid. What do you prefer?

Reply:
I'm fine with ascii, I don't know of any good reasons to use non-ascii whitespace in the configuration file.

Reply:
I don't think this has to do with non-ascii whitespace. Currently, we accept anything that is not a separator for a string. This might be too liberal, as it includes ascii control characters, the null character, and more. We should probably be more selective in what we accept.